CUDADataFormats/Track/interface/TrajectoryStateSoAT.h

```diff
@@ -25,12 +25,12 @@ struct TrajectoryStateSoAT {
     auto cov = covariance(i);
     cov(0) = ccov(0, 0);
     cov(1) = ccov(0, 1);
-    cov(2) = b * float(ccov(0, 2));
+    cov(2) = b * static_cast<float>(ccov(0, 2));
```

Comment:
sorry, do not see the reason to add static_cast

Reply:
This is purely switching to C++ syntax for the cast.

Reply:
just adding useless characters. I claim that for intrinsic type "C-type cast" is more readable and more appropriate.
There is NO other type of cast that can be applied here. The semantic MUST be the of "C-type cast". Not of any C++ cast.

Reply:
Is there any use case where it should translate to a `reinterpret_cast`? There is no dynamic cast for sure with intrisic types and no const correctness issue with an rvalue.

Reply:
Sorry, reinterpret_cast between double and float is invalid.

Reply:
> just adding useless characters. I claim that for intrinsic type "C-type cast" is more readable and more appropriate.
> There is NO other type of cast that can be applied here. The semantic MUST be the of "C-type cast". Not of any C++ cast.

I propose to get an exception in the rule clarified after some discussion in the core group.
@cms-sw/core-l2

Reply:
The rule 4.15 already allows exceptions, and admittedly explicit narrowing conversions within floating point types falls on the low-risk side of C-style casts.

The C++ Core Guidelines advocates for [`gsl::narrow_cast`](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es46-avoid-lossy-narrowing-truncating-arithmetic-conversions) (from the guidelines support library). While that describes the intent well, it wouldn't help towards "adding useless characters".